test/v1/connectors/DeleteCharitableGivingTaxReliefConnectorSpec.scala

```diff
@@ -26,35 +26,48 @@ import scala.concurrent.Future
 
 class DeleteCharitableGivingTaxReliefConnectorSpec extends ConnectorSpec {
 
-  val nino: String    = "AA123456A"
-  val taxYear: String = "2019-20"
+  private val nino = "AA123456A"
 
-  trait Test { _: ConnectorTest =>
-
-    val connector: DeleteCharitableGivingTaxReliefConnector = new DeleteCharitableGivingTaxReliefConnector(
-      http = mockHttpClient,
-      appConfig = mockAppConfig
-    )
+  "delete()" should {
+    "return a success response" when {
+      "given a non-TYS request" in new DesTest with Test {
 
-  }
+        willPost(
+          url = s"$baseUrl/income-tax/nino/$nino/income-source/charity/annual/2020",
+          body = JsObject.empty
+        )
+          .returns(Future.successful(expectedOutcome))
 
-  "delete" should {
+        private val request = DeleteCharitableGivingTaxReliefRequest(Nino(nino), TaxYear.fromMtd("2019-20"))
+        private val result  = await(connector.delete(request))
 
-    val request: DeleteCharitableGivingTaxReliefRequest = DeleteCharitableGivingTaxReliefRequest(Nino(nino), TaxYear.fromMtd(taxYear))
+        result shouldBe expectedOutcome
+      }
 
-    "return a result" when {
-      "the downstream call is successful" in new DesTest with Test {
-        val outcome = Right(ResponseWrapper(correlationId, ()))
+      "given a TYS request" in new TysIfsTest with Test {
 
-        willPost(
-          url = s"$baseUrl/income-tax/nino/$nino/income-source/charity/annual/${request.taxYear.asDownstream}",
-          body = JsObject.empty
+        willDelete(
+          url = s"$baseUrl/income-tax/23-24/$nino/income-source/charity/annual"
         )
-          .returns(Future.successful(outcome))
+          .returns(Future.successful(expectedOutcome))
+
+        private val request = DeleteCharitableGivingTaxReliefRequest(Nino(nino), TaxYear.fromMtd("2023-24"))
+        private val result  = await(connector.delete(request))
 
-        await(connector.delete(request)) shouldBe outcome
+        result shouldBe expectedOutcome
       }
     }
   }
 
+  trait Test { _: ConnectorTest =>
+
+    protected val expectedOutcome = Right(ResponseWrapper(correlationId, ()))
+
+    protected val connector: DeleteCharitableGivingTaxReliefConnector = new DeleteCharitableGivingTaxReliefConnector(
+      http = mockHttpClient,
+      appConfig = mockAppConfig
```

Comment:
Could we abstract out the request from the test and move a request val into the Test trait in something like:
`
def taxYear: TaxYear

private val request = DeleteCharitableGivingTaxReliefRequest(Nino(nino), taxYear)
`

though this would then mean having to define the taxYear in each test

Reply:
I went that way initially, it ended up with more boilerplate code/complexity; sometimes better to have a little bit of duplication to avoid that...